diff --git a/src/luajit-gdb.py b/src/luajit-gdb.py
index d2070e9b..05351b9f 100644
--- a/src/luajit-gdb.py
+++ b/src/luajit-gdb.py
@@ -327,7 +327,8 @@ def itypemap(o):
 
 
 def funcproto(func):
-    assert func['ffid'] == 0
+    if not (func['ffid'] == 0):
+        raise RuntimeError("assert failed")
 
     return cast('GCproto *',
                 mref('char *', func['pc']) - gdb.lookup_type('GCproto').sizeof)
