diff --git a/src/lib/core/say.c b/src/lib/core/say.c
index a80880724..2ab3d5478 100644
--- a/src/lib/core/say.c
+++ b/src/lib/core/say.c
@@ -655,6 +655,7 @@ log_syslog_init(struct log *log, const char *init_str)
 	else
 		log->syslog_ident = strdup(opts.identity);
 	if (log->syslog_ident == NULL) {
+		assert(opts.identity != NULL);
 		diag_set(OutOfMemory, strlen(opts.identity), "malloc",
 		         "log->syslog_ident");
 		return -1;
