diff --git a/src/lib/salad/bps_tree.h b/src/lib/salad/bps_tree.h
index e340588cb..7cdeb9770 100644
--- a/src/lib/salad/bps_tree.h
+++ b/src/lib/salad/bps_tree.h
@@ -1585,6 +1585,7 @@ bps_tree_random(const struct bps_tree *t, size_t rnd)
 	for (bps_tree_block_id_t i = 0; i < tree->depth - 1; i++) {
 		struct bps_inner *inner = (struct bps_inner *)block;
 		bps_tree_pos_t pos = rnd % inner->header.size;
+		assert(pos < BPS_TREE_MAX_COUNT_IN_INNER);
 		rnd /= inner->header.size;
 		block = bps_tree_restore_block(tree, inner->child_ids[pos]);
 	}
