diff --git a/src/box/request.c b/src/box/request.c
index 4c923c6ac..dacb9e7de 100644
--- a/src/box/request.c
+++ b/src/box/request.c
@@ -187,8 +187,9 @@ request_handle_sequence(struct request *request, struct space *space,
 	}
 
 	if (path != NULL) {
-		tuple_go_to_path(&key, path, strlen(path), TUPLE_INDEX_BASE,
-				 MULTIKEY_NONE);
+		if (unlikely(tuple_go_to_path(&key, path, strlen(path),
+			TUPLE_INDEX_BASE, MULTIKEY_NONE) != 0))
+			return -1;
 		if (key == NULL)
 			return 0; /* field not found */
 	}
