diff --git a/src/box/vinyl.c b/src/box/vinyl.c
index 3c6311f75..4002853ee 100644
--- a/src/box/vinyl.c
+++ b/src/box/vinyl.c
@@ -627,6 +627,10 @@ vinyl_engine_create_space(struct engine *engine, struct space_def *def,
 		keys = xregion_alloc_array(&fiber()->gc, typeof(keys[0]),
 					   key_count);
 	key_count = 0;
+	if (keys == NULL) {
+		free(space);
+		return NULL;
+	}
 	rlist_foreach_entry(index_def, key_list, link)
 		keys[key_count++] = index_def->key_def;
 
