diff --git a/src/box/alter.cc b/src/box/alter.cc
index 7aeab4648..7b8cd489c 100644
--- a/src/box/alter.cc
+++ b/src/box/alter.cc
@@ -1774,6 +1774,7 @@ alter_space_move_indexes(struct alter_space *alter, uint32_t begin,
 					old_def->name, strlen(old_def->name),
 					old_def->type, &old_def->opts,
 					old_def->key_def, alter->pk_def);
+		assert(new_def != NULL);
 		index_def_update_optionality(new_def, min_field_count);
 		auto guard = make_scoped_guard([=] { index_def_delete(new_def); });
 		if (!index_def_change_requires_rebuild(old_index, new_def))
@@ -2996,6 +2997,8 @@ on_replace_dd_user(struct trigger * /* trigger */, void *event)
 			return -1;
 		txn_stmt_on_rollback(stmt, on_rollback);
 	} else if (new_tuple == NULL) { /* DELETE */
+		if (old_user == NULL)
+			return -1; /* using if instead of assert for svace */
 		if (access_check_ddl(old_user->def->name, old_user->def->uid,
 				 old_user->def->owner, old_user->def->type,
 				 BOX_PRIVILEGE_DROP) != 0)
@@ -3353,6 +3356,8 @@ on_replace_dd_func(struct trigger * /* trigger */, void *event)
 		if (trigger_run(&on_alter_func, func) != 0)
 			return -1;
 	} else if (new_tuple == NULL) {         /* DELETE */
+		if (old_func == NULL)
+			return -1; /* using if instead of assert for svace */
 		uint32_t uid;
 		if (func_def_get_ids_from_tuple(old_tuple, &fid, &uid) != 0)
 			return -1;
@@ -4392,7 +4397,9 @@ on_replace_dd_sequence(struct trigger * /* trigger */, void *event)
 		sequence_cache_delete(seq->def->id);
 		txn_stmt_on_commit(stmt, on_commit);
 		txn_stmt_on_rollback(stmt, on_rollback);
-	} else {						/* UPDATE */
+	} else {
+		if (new_tuple == NULL)			/* UPDATE */
+			return -1; /* using if instead of assert for svace */
 		new_def = sequence_def_new_from_tuple(new_tuple,
 						      ER_ALTER_SEQUENCE);
 		if (new_def == NULL)
