diff --git a/src/httpc.c b/src/httpc.c
index 14880200d..fd351e921 100644
--- a/src/httpc.c
+++ b/src/httpc.c
@@ -559,6 +559,10 @@ httpc_request_io_read(struct httpc_request *req, char *buf, size_t len,
 		if (recv_len > remain) {
 			const size_t tocopy = recv_len - remain;
 			char *ptr = ibuf_alloc(&req->io_recv, tocopy);
+			if (ptr == NULL) {
+				diag_set(OutOfMemory, len, "ibuf", "alloc");
+				return -1;
+			}
 			memcpy(ptr, recv + remain, tocopy);
 		}
 
@@ -607,6 +611,11 @@ httpc_request_io_write(struct httpc_request *req, const char *ptr, size_t len,
 	if (len > 0) {
 		ibuf_reset(&req->send);
 		char *buf = ibuf_alloc(&req->send, len);
+		if (buf == NULL){
+			diag_set(OutOfMemory, len, "ibuf", "alloc");
+			return -1;
+		}
+
 		memcpy(buf, ptr, len);
 	} else {
 		req->io_send_closed = true;
