diff --git a/src/box/memtx_tx.c b/src/box/memtx_tx.c
index df34ef32d..b41938521 100644
--- a/src/box/memtx_tx.c
+++ b/src/box/memtx_tx.c
@@ -3192,7 +3192,8 @@ memtx_tx_nearby_gap_item_new(struct txn *txn, enum iterator_type type,
 		item->key = memtx_tx_xregion_alloc(txn, item->key_len,
 						   MEMTX_TX_ALLOC_TRACKER);
 	}
-	memcpy((char *)item->key, key, item->key_len);
+	if (item->key && key && item->key_len)
+		memcpy((char *)item->key, key, item->key_len);
 	return item;
 }
 
