diff --git a/src/lib/core/say.c b/src/lib/core/say.c
index 715bf373b..e2b210721 100644
--- a/src/lib/core/say.c
+++ b/src/lib/core/say.c
@@ -655,7 +655,10 @@ log_syslog_init(struct log *log, const char *init_str)
 	else
 		log->syslog_ident = strdup(opts.identity);
 	if (log->syslog_ident == NULL) {
-		diag_set(OutOfMemory, strlen(opts.identity), "malloc",
+		int len = 0;
+		if (opts.identity != NULL)
+			len = strlen(opts.identity);
+		diag_set(OutOfMemory, len, "malloc",
 		         "log->syslog_ident");
 		return -1;
 	}
