diff --git a/src/tool_paramhlp.c b/src/tool_paramhlp.c
index d4024e134..efe11ee9a 100644
--- a/src/tool_paramhlp.c
+++ b/src/tool_paramhlp.c
@@ -397,9 +397,10 @@ ParameterError proto2num(struct OperationConfig *config,
 
   /* Allow strtok() here since this is not used threaded */
   /* !checksrc! disable BANNEDFUNC 2 */
-  for(token = strtok(buffer, sep);
+  char *saveptr = NULL;
+  for(token = strtok_r(buffer, sep, &saveptr);
       token;
-      token = strtok(NULL, sep)) {
+      token = strtok_r(NULL, sep, &saveptr)) {
     enum e_action { allow, deny, set } action = allow;
 
     /* Process token modifiers */
