diff --git a/third_party/lua-cjson/strbuf.c b/third_party/lua-cjson/strbuf.c
index 69057ffb3..5a0272be1 100644
--- a/third_party/lua-cjson/strbuf.c
+++ b/third_party/lua-cjson/strbuf.c
@@ -106,7 +106,7 @@ static int calculate_new_size(strbuf_t *s, int len)
         /* Exponential sizing */
         while (newsize < reqsize)
             newsize *= -s->increment;
-    } else {
+    } else if (s->increment > 0) {
         /* Linear sizing */
         newsize = ((newsize + s->increment - 1) / s->increment) * s->increment;
     }
