diff --git a/src/box/key_def.c b/src/box/key_def.c
index c790e2fb1..af9b4dd16 100644
--- a/src/box/key_def.c
+++ b/src/box/key_def.c
@@ -219,7 +219,9 @@ key_def_set_part_path(struct key_def *def, uint32_t part_no, const char *path,
 	struct json_token token;
 	json_lexer_create(&lexer, path + multikey_path_len,
 			  path_len - multikey_path_len, TUPLE_INDEX_BASE);
-	json_lexer_next_token(&lexer, &token);
+	int rc = json_lexer_next_token(&lexer, &token);
+	if (rc != 0) // assert
+		panic("rc != 0 in %s:%d", __FILE__, __LINE__);
 	assert(token.type == JSON_TOKEN_ANY);
 
 	/* The rest of JSON path couldn't be multikey. */
