From f30c339585525b7eebc1714b3a5dd263723177f8 Mon Sep 17 00:00:00 2001
From: Georgy Moshkin <gmoshkin@picodata.io>
Date: Thu, 24 Feb 2022 06:43:20 -0800
Subject: [PATCH] build: update openssl version to support arm64-apple

---
 static-build/CMakeLists.txt |  8 ++++----
 6 files changed, 51 insertions(+), 10 deletions(-)

diff --git a/static-build/CMakeLists.txt b/static-build/CMakeLists.txt
index 598c42bbc..2ff8906b1 100644
--- a/static-build/CMakeLists.txt
+++ b/static-build/CMakeLists.txt
@@ -8,11 +8,11 @@ cmake_minimum_required(VERSION 2.8)
 project(tarantool-static C CXX)
 
 include(ExternalProject)
-set(OPENSSL_VERSION 1.1.1f)
-set(OPENSSL_HASH 3f486f2f4435ef14b81814dbbc7b48bb)
+set(OPENSSL_VERSION 1.1.1m)
+set(OPENSSL_HASH 8ec70f665c145c3103f6e330f538a9db)
 set(ZLIB_VERSION 1.2.11)
 set(ZLIB_HASH 1c9f62f0778697a09d36121ead88e08e)
 set(NCURSES_VERSION 6.2)
 set(NCURSES_HASH e812da327b1c2214ac1aed440ea3ae8d)
 set(READLINE_VERSION 8.0)
 set(READLINE_HASH 7e6c1f16aee3244a69aba6e438295ca3)
-- 
2.32.0 (Apple Git-132)

