From da9f7b4ee2423fb04976efca9f253d8c87e18002 Mon Sep 17 00:00:00 2001
From: Georgy Moshkin <gmoshkin@picodata.io>
Date: Tue, 8 Mar 2022 10:47:31 -0800
Subject: [PATCH] build: fix build on macOS 11.5 or higher

---
 static-build/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/static-build/CMakeLists.txt b/static-build/CMakeLists.txt
index 598c42bbc..3525e294c 100644
--- a/static-build/CMakeLists.txt
+++ b/static-build/CMakeLists.txt
@@ -205,7 +205,7 @@ if (APPLE)
     # So we need to find library and header and
     # copy it locally
     find_path(UNWIND_INCLUDE_DIR libunwind.h)
-    find_library(UNWIND_LIBRARY libunwind.dylib
+    find_library(UNWIND_LIBRARY libunwind.tbd
         PATH_SUFFIXES system
     )
 
@@ -225,7 +225,7 @@ if (APPLE)
                 "${UNWIND_INSTALL_PREFIX}/include/"
         )
         add_custom_command(
-            OUTPUT "${UNWIND_INSTALL_PREFIX}/lib/libunwind.dylib"
+            OUTPUT "${UNWIND_INSTALL_PREFIX}/lib/libunwind.tbd"
             COMMAND ${CMAKE_COMMAND} -E make_directory
                 "${UNWIND_INSTALL_PREFIX}/lib"
             COMMAND ${CMAKE_COMMAND} -E copy
@@ -234,7 +234,7 @@ if (APPLE)
         )
         set(UNWIND_DEPENDENCIES
             ${UNWIND_DEPENDENCIES}
-            "${UNWIND_INSTALL_PREFIX}/lib/libunwind.dylib"
+            "${UNWIND_INSTALL_PREFIX}/lib/libunwind.tbd"
             "${UNWIND_INSTALL_PREFIX}/include/libunwind.h"
         )
     else()
-- 
2.32.0 (Apple Git-132)

