From cfbf0fadf54c6d328e4e05974c140a715c197705 Mon Sep 17 00:00:00 2001
From: Georgy Moshkin <gmoshkin@picodata.io>
Date: Thu, 14 Jul 2022 17:29:58 +0300
Subject: [PATCH] build: vendor the dependencies

---
 static-build/CMakeLists.txt | 30 ++++++------------------------
 1 file changed, 6 insertions(+), 24 deletions(-)

diff --git a/static-build/CMakeLists.txt b/static-build/CMakeLists.txt
index c830daf01..62417fdc2 100644
--- a/static-build/CMakeLists.txt
+++ b/static-build/CMakeLists.txt
@@ -8,18 +8,6 @@ cmake_minimum_required(VERSION 2.8)
 project(tarantool-static C CXX)
 
 include(ExternalProject)
-set(LIBICU_VERSION release-71-1/icu4c-71_1)
-set(LIBICU_HASH e06ffc96f59762bd3c929b217445aaec)
-set(LIBICONV_VERSION 1.16)
-set(LIBICONV_HASH 7d2a800b952942bb2880efb00cfd524c)
-set(OPENSSL_VERSION 1.1.1n)
-set(OPENSSL_HASH 2aad5635f9bb338bc2c6b7d19cbc9676)
-set(ZLIB_VERSION 1.2.11)
-set(ZLIB_HASH 1c9f62f0778697a09d36121ead88e08e)
-set(NCURSES_VERSION 6.2)
-set(NCURSES_HASH e812da327b1c2214ac1aed440ea3ae8d)
-set(READLINE_VERSION 8.0)
-set(READLINE_HASH 7e6c1f16aee3244a69aba6e438295ca3)
 set(BACKUP_STORAGE https://distrib.hb.bizmrg.com)
 
 # Pass -isysroot=<SDK_PATH> option on Mac OS to a preprocessor and a C
@@ -47,8 +35,7 @@ endif()
 # OpenSSL
 #
 ExternalProject_Add(openssl
-    URL ${BACKUP_STORAGE}/openssl/openssl-${OPENSSL_VERSION}.tar.gz
-    URL_MD5 ${OPENSSL_HASH}
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../vendor/openssl-1.1.1m
     CONFIGURE_COMMAND <SOURCE_DIR>/config
         CC=${CMAKE_C_COMPILER}
         CXX=${CMAKE_CXX_COMPILER}
@@ -66,8 +53,7 @@ ExternalProject_Add(openssl
 # ICU
 #
 ExternalProject_Add(icu
-    URL https://github.com/unicode-org/icu/releases/download/${LIBICU_VERSION}-src.tgz
-    URL_MD5 ${LIBICU_HASH}
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../vendor/icu4c-62_1
     # By default libicu is built by using clang/clang++ compiler (if it
     # exists). Here is a link for detecting compilers at libicu configure
     # script: https://github.com/unicode-org/icu/blob/7c7b8bd5702310b972f888299169bc3cc88bf0a6/icu4c/source/configure.ac#L135
@@ -93,8 +79,7 @@ ExternalProject_Add(icu
 # ZLIB
 #
 ExternalProject_Add(zlib
-    URL ${BACKUP_STORAGE}/zlib/zlib-${ZLIB_VERSION}.tar.gz
-    URL_MD5 ${ZLIB_HASH}
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../vendor/zlib-1.2.11
     CONFIGURE_COMMAND env
         CC=${CMAKE_C_COMPILER}
         CFLAGS=${DEPENDENCY_CFLAGS}
@@ -109,8 +94,7 @@ ExternalProject_Add(zlib
 # Ncurses
 #
 ExternalProject_Add(ncurses
-    URL https://ftp.gnu.org/gnu/ncurses/ncurses-${NCURSES_VERSION}.tar.gz
-    URL_MD5 ${NCURSES_HASH}
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../vendor/ncurses-6.2
     CONFIGURE_COMMAND <SOURCE_DIR>/configure
         CC=${CMAKE_C_COMPILER}
         CXX=${CMAKE_CXX_COMPILER}
@@ -143,8 +127,7 @@ ExternalProject_Add(ncurses
 # ReadLine
 #
 ExternalProject_Add(readline
-    URL https://ftp.gnu.org/gnu/readline/readline-${READLINE_VERSION}.tar.gz
-    URL_MD5 ${READLINE_HASH}
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../vendor/readline-8.0
     CONFIGURE_COMMAND <SOURCE_DIR>/configure
         CC=${CMAKE_C_COMPILER}
         CFLAGS=${DEPENDENCY_CFLAGS}
@@ -160,8 +143,7 @@ ExternalProject_Add(readline
 #
 if (APPLE)
     ExternalProject_Add(iconv
-        URL https://ftp.gnu.org/pub/gnu/libiconv/libiconv-${LIBICONV_VERSION}.tar.gz
-        URL_MD5 ${LIBICONV_HASH}
+        SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../vendor/libiconv-1.16
         CONFIGURE_COMMAND <SOURCE_DIR>/configure
             CC=${CMAKE_C_COMPILER}
             CFLAGS=${DEPENDENCY_CFLAGS}
-- 
2.25.1

