From d3b1b69d32ebef1ec229fe11730d73fce9880516 Mon Sep 17 00:00:00 2001
From: Georgy Moshkin <gmoshkin@picodata.io>
Date: Thu, 10 Mar 2022 20:36:36 +0300
Subject: [PATCH] build: remove some unneeded build steps for ncurses

---
 static-build/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/static-build/CMakeLists.txt b/static-build/CMakeLists.txt
index 598c42bbc..d3484c9be 100644
--- a/static-build/CMakeLists.txt
+++ b/static-build/CMakeLists.txt
@@ -133,6 +133,9 @@ ExternalProject_Add(ncurses
         --disable-db-install
         --without-progs
         --without-manpages
+        --without-manpage-renames
+        --without-ada
+        --without-tests
 )
 
 #
-- 
2.25.1

